### ISSUE TITLE
Data type error in cudnn.SpatialConvolution

### ISSUE THREAD

**Issue by crazylyf** [soumith/cudnn.torch#304]
Hi all,

I have found an very strange error: when I train my model (contains cudnn.SpatialConvolution), it throw an error: "unable to convert argument 7 from cdata<long long*> to cdata<int*>". The error can be fixed by changing the bufSize (line 378) from LongTensor to IntTensor. However, when I try a  simple example, it goes well only when bufSize is LongTensor! The example code are as follows:

require('nn')
require('cudnn')

local dtype = 'torch.CudaTensor'
local data = torch.FloatTensor(5, 3, 10, 10):type(dtype)
local model = nn.Sequential():add(cudnn.SpatialConvolution(3, 5, 3, 3, 1, 1, 1, 1)):type(dtype)
local output = model:forward(data)
print(output)

Can anyone help me find out the exact reason? thanks!

**Comment by BTNC** [comment]
What platform are you on? The underlining api is expecting a long long pointer to write to, you can not use a int pointer since it will cause writing 64 bit integer to the 32 bit integer address. Try to replace LongTensor with a 64 bit integer type with the help of ffi.

**Comment by crazylyf** [comment]
The script run on "CentOS Linux release 7.2.1511 (Core)" with lua5.2 and cuda-7.5.
I will try ffi to solve the problem as you pointed out. thanks @BTNC

**Comment by antonheryanto** [comment]
encounter similar problem when running https://github.com/szagoruyko/wide-residual-networks train.lua
run on "Win 10" with C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\bin\cudnn64_5.dll

install\luarocks\systree/share/lua/5.1/cudnn\find.lua:394: bad argument #7 to 'call' (cannot convert 'int *' to 'uint64_t *')

from thread above using ffi to replace long tensor, but because i didn't know exactly how to do it, just try and errors as my skill limited

find.lua:394
```lua
-- local bufSize = ffi.new("uint64_t") -- fail
-- local bufSize = ffi.new("uint64_t[0]")  -- fail
local bufSize = ffi.new("uint64_t *")  -- fail
ret = cudnn.call(getWSAlgos[findAPI_idx],
                         cudnn.getHandle(),
                         params[1], params[3], layer.convDesc[0], params[6],
                         retAlgo, bufferSize)
```
this successful
```lua
local bufSize = torch.LongTensor(1)
ret = cudnn.call(getWSAlgos[findAPI_idx],
                         cudnn.getHandle(),
                         params[1], params[3], layer.convDesc[0], params[6],
                         retAlgo, ffi.cast("uint64_t *", bufSize:data()))
```
just didn't sure whether this correct approach, any suggestion welcome

**Comment by BTNC** [comment]
Hi @antonheryanto , torch cudnn package has been patched to replace LongTensor by [99bc14e](https://github.com/soumith/cudnn.torch/commit/99bc14eac2e3653adb87348e28d39ec1f40012b0) and [8903d1b](https://github.com/soumith/cudnn.torch/commit/8903d1b389e49175de0493e9507306c7be89612f), you can reinstall latest torch cudnn package.

**Comment by antonheryanto** [comment]
@BTNC i was using your torch [distro](https://github.com/BTNC/distro-win) pulling latest changes and reinstall worked for me. thank you very much
